packages/shared/scope-hoisting/src/link.js

```diff
@@ -334,7 +346,18 @@ export function link({
       invariant(imported.assets != null);
       imported.assets.add(mod);
 
-      return t.callExpression(getIdentifier(mod, 'init'), []);
+      let initIdentifier = getIdentifier(mod, 'init');
+
+      let program = path.scope.getProgramParent().path;
+      if (!program.scope.hasOwnBinding(initIdentifier.name)) {
+        // add binding so we can track the scope
+        let [decl] = program.unshiftContainer('body', [
+          t.variableDeclaration('var', [t.variableDeclarator(initIdentifier)]),
+        ]);
+        program.scope.registerDeclaration(decl);
+      }
+
+      return t.callExpression(initIdentifier, []);
```

Comment:
reference init identifier?

Reply:
This PR only applies to `shake` and the output-formats functions. (This is also why there is still a crawl call in link in `Program.exit`)

> I plan on removing the last call in link.js as well in a follow up, but this PR should contain most of the performance benefits and I want to avoid rebasing this once again.